src/VisualStudio/VisualStudioDiagnosticsToolWindow/Telemetry/TelemetryPanel.xaml.cs

```diff
@@ -27,20 +27,24 @@ public TelemetryPanel()
             InitializeComponent();
         }
 
-#pragma warning disable VSTHRD100 // Avoid async void methods
-        private async void OnDump(object sender, RoutedEventArgs e)
-#pragma warning restore VSTHRD100 // Avoid async void methods
+        private void OnDump(object sender, RoutedEventArgs e)
         {
-            using (Disable(DumpButton))
-            using (Disable(CopyButton))
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
Did you mean this to be RunAsync or something? Otherwise this is blocking whatever thread it's on when we hop to the bcakground thread?

Reply:
Yes this one can be `RunAsync`.

Reply:
Fixed!